Feature/promo search

do-kevin opened the pull request:
https://github.com/keburke17/Evergreen-api/pull/38

Commit:
Add Promo

do-kevin committed:
Adds Checkboxes for promo exclusive to admin role

do-kevin committed:
Adds Promotions page for admins

do-kevin committed:
Adds PromoteCard which toggles promo statuses

do-kevin committed:
Adds filtering for PromotionContainer

do-kevin committed:
Displays main and local promos in carousels

do-kevin committed:
Shows search results in SearchResultContainer

lan-nguyen91 commented on src/components/SearchResultContainer/SearchResultContainer.js:
```diff
@@ -0,0 +1,127 @@
+
+  return (
+    <div className="h-full">
+      {data && data.length && <p className="text-2xl text-center">Results</p>}
```
move this up to a early return

if (!data.length) return some kind of empty data logo

lan-nguyen91 commented on src/components/SearchResultContainer/SearchResultContainer.js:
```diff
@@ -0,0 +1,127 @@
+        groupKeys.map((key) => {
+          return (
+            <div key={key}>
+              {(group[key] && group[key].length && (
```
move this to a component some where in this file, don't keep using ternary

lan-nguyen91 replied:
just call something like 

<FilterredGroup group={group} />

lan-nguyen91 commented on src/components/promotion/PromotionContainer.js:
```diff
@@ -0,0 +1,210 @@
+
+  if (showPromoted) {
+    showData = showData.filter((d) => {
+      if (
```
im unsure what is the data look like but this can be simplify for sure, can you paste date format

do-kevin committed:
Adds filters for searchbox in HomeScreen

Can filter results by topic

do-kevin committed:
Changes offer enrollment endpoint; also includes provider id

do-kevin committed:
Cleans up useEffect and rerendering issue for Offer info layout

Cleans up useEffect for pathway info layout

Cleans up useEffects for Topic carousel and provider info layout

do-kevin committed:
Fix minor bugs and sets new enrollment endpoint

do-kevin committed:
Change style and makes InfoCard clickable

Tweaks styles for small info card

Removes useEffect and tweaks style for promo carousels
